File: Laborator4/Laborator4_RecycleView2/app/src/main/java/com/example/laborator4_recycleview2/MainActivity.java
package com.example.laborator4_recycleview2;


import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    RecyclerView recyclerView;
    ProductAdapter adapter;
    List<Product> productList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerViewProducts);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        productList = new ArrayList<>();
        productList.add(new Product("Laptop Lenovo", "3.500 RON", R.drawable.laptop));
        productList.add(new Product("Telefon Samsung", "2.400 RON", R.drawable.phone));
        productList.add(new Product("Căști Bluetooth", "350 RON", R.drawable.headphones));
        productList.add(new Product("Smartwatch Huawei", "899 RON", R.drawable.smrtw));

        adapter = new ProductAdapter(this, productList);
        recyclerView.setAdapter(adapter);
    }
}

File: Demos/RecyclerViewDemo/app/src/main/java/com/example/demo/MainActivity.kt
package com.example.demo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var itemList: ArrayList<Item>
    private lateinit var itemAdapter: ItemAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // Setăm layout-ul activității
        setContentView(R.layout.activity_main)

        // Inițializăm RecyclerView
        recyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.setHasFixedSize(true)

        // Inițializăm lista de date
        itemList = ArrayList()
        for (i in 1..50) {
            itemList.add(Item("Element $i"))
        }

        // Setăm adapterul
        itemAdapter = ItemAdapter(itemList)
        recyclerView.adapter = itemAdapter
    }
}
